File: docs/snippet/code-signal/arcade/intro/smooth-sailing/reverse-in-parentheses/reverse-in-parentheses.py
import re


def reverse_in_parentheses(inputString):
    m = re.search(r"\(([^()]*)\)", inputString)
    if m is None:
        return inputString
    return reverse_in_parentheses(inputString[: m.start()] + m.group(1)[::-1] + inputString[m.end():])


print(reverse_in_parentheses("(bar)"))

File: content/snippet/code-signal/arcade/intro/smooth-sailing/reverse-in-parentheses/reverse-in-parentheses.js
function reverseInParentheses(inputString) {
  const regex = /\(([^()]*)\)/;
  let result = inputString;
  let match = result.match(regex);

  while (match) {
    const reversed = match[1].split('').reverse().join('');
    result = result.replace(regex, reversed);
    match = result.match(regex);
  }

  return result;
}

console.log(reverseInParentheses('foo(bar(baz))blim'));
